nucypher/blockchain/eth/interfaces.py

```diff
@@ -311,6 +328,7 @@ def connect(self):
         # Connect if not connected
         try:
             self.w3 = self.Web3(provider=self._provider)
+            self.tx_machine.w3 = self.w3  # share this web3 instance with the tracker
```

Comment:
This seems a bit hacky. Also, wasn't this already done in L222?

Reply:
Good eye 😅 

> wasn't this already done in L222?

The `Configuration` object lifecycle includes a stage where `self.w3 is NO_BLOCKCHAIN_CONNECTION` (L219) until `.connect` is called, where the w3 instance is actually resolved.

This line is mimicking the same "connection" behavior as the interface itself.